event-processing/src/main/java/com/camunda/consulting/eventprocessing/EventRepository.java

```diff
@@ -0,0 +1,85 @@
+package com.camunda.consulting.eventprocessing;
+
+import com.camunda.consulting.eventprocessing.EventRepository.EventEntity;
+import jakarta.persistence.Entity;
+import jakarta.persistence.Id;
+import java.time.OffsetDateTime;
+import org.springframework.data.jpa.repository.JpaRepository;
+import org.springframework.stereotype.Repository;
+
+@Repository
+public interface EventRepository extends JpaRepository<EventEntity, String> {
+
+  @Entity(name = "EVENT")
+  class EventEntity {
```

Comment:
In modern Java, this could be a `record` instead of a `class`. Does it work with Spring Data?

Reply:
I see that records are used in the service.

Reply:
I know, I tried to a make it work, but I failed... so I decided to use the old-school verbose pojo class

Reply:
OK, reason found here: https://www.baeldung.com/spring-jpa-java-records#1-records-cant-be-entities